oap-server/analyzer/log-analyzer/src/main/java/org/apache/skywalking/oap/log/analyzer/dsl/DSL.java

```diff
@@ -53,6 +62,22 @@ public static DSL of(final ModuleManager moduleManager,
                 CompileStatic.class
             );
         cc.addCompilationCustomizers(customizer);
+        final SecureASTCustomizer secureASTCustomizer = new SecureASTCustomizer();
+        secureASTCustomizer.setDisallowedStatements(
+            ImmutableList.<Class<? extends Statement>>builder()
+                         .add(WhileStatement.class)
+                         .add(DoWhileStatement.class)
+                         .add(ForStatement.class)
```

Comment:
Do we need a `If`?

Reply:
> Do we need a `If`?

We need `if` statement, note this is `DisallowedStatements`

Reply:
Oh, I see.